packages/core/components/icon/index.js

```diff
@@ -24,19 +24,18 @@ const StyledIcon = styled.svg`
   ${POSITION}
 `;
 
-const Icon = ({ ...p }) => {
+const Icon = ({ path, ...p }) => {
```

Comment:
Awesome. Just realized you should extract `viewBox` here, too, so it doesn't get passed automatically to `StyledIcon` in line 36, potentially overriding the logic in line 34.

Reply:
I don't think this is necessary. Styled-components handles merging properties. With the ternary operation on line 34, sc shouldn't run into any issues with merging properties since they would always be identical. 

That being said, I'm not an expert by any means, so if you still think I should still make the change I'm happy to do it.

Reply:
I guess the case I'm worried about is if the user passes both a `name` prop and a `viewBox` prop. In that case, line 34 evaluates to `viewBox={icon.viewBox}` while line 36 generates `viewBox={p.viewBox}`. This results in two `viewBox` props, with the second one winning. That said, the outcome is probably the correct outcome, ie, the viewBox manually passed in overrides the one baked into the icon, so… yeah assuming you agree, then you can ignore. :)

Reply:
That case you described will actually always have the name's viewBox (set in the theme) take precedence, which is the intended behavior. I did this purposefully so that people don't unintentionally break an icon defined in the theme. 

My intent was such that `path` and `viewbox` would always be paired and could not break/interfere with already defined icons in the theme.

As for your initial suggestion, I think to be on the safe side of things, I'll make your suggested change. It doesn't change how things function and if anything, just adds another layer of safety.